architecture/2025-02-user-defined-resource-type-feature-spec.md

```diff
@@ -0,0 +1,1303 @@
+    version: 'v1alpha1'
+    schema: {
+      properties: {
++       // Required input property
+        size: {
+          type: 'string'
+          enum: ['S', 'M', 'L', 'XL']
+        }
++        // Optional input property
+        logging-verbosity: {
+          type: string
+          enum: ['TERSE', 'DEFAULT', 'VERBOSE']
+        }
++        // Output property set by the recipe, note the readOnly property
+        connection-string: {
+          type: 'string'
++          readOnly: true
+        }
++        // Output property set by the recipe, note the readOnly property
+        credentials: {
+          type: 'object'
++          readOnly: true
+          properties: {
+            username: {
+              type: 'string'
+            }
+            password: {
+              type: 'string'
+            }
+          }
+        }
++     // Specifying required properties
++      required: ['size']
+    }
+  }
+}
+```
+
+### User Story 3 – Reading Properties
+
+As a developer, I need to read properties for resources in my application definition and set environment variables for my container. For example, when I create a database resource, I need to set an environment variable in my container which gives my application the connection string.
+
+**User Experience 1 – Set manually via Bicep by the developer**
+
+The develop can manually inject environment variables into the container using Bicep. 
+
+> [!NOTE]
+>
+> This functionality exists today. There are no changes. It is included here for context with the other examples.
+
+**`my-backend-application.bicep`**:
+
+```diff
+resource ordersDB 'MyCompany.Data/postgreSQL@v1alpha1' = {
+  name: 'ordersDB'
+  properties: {
+     size: 'M' 
+  }
+}
+
+resource backend 'Applications.Core/containers@2023-10-01-preview' = {
+  name: 'backend'
+  properties: {
+    container: {
+      image: 'my-applicatio-container:latest'
+      connections: {
+      ordersDB: {
+        source: ordersDB.id
+      }
++      // Set environment variables in the container by referencing properties via Bicep
++      env:{
++        ORDERS_DB_CONNECTION_STRING: {
++          value: ordersDB.connectionString
++        }
++        ORDERS_DB_USERNAME: {
++          value: ordersDB.credentias.username
++        }
++        ORDERS_DB_PASSWORD: {
++          value: ordersDB.credentias.password
++        }
+      }
+   }
+}
+```
+
+**Result**
+
+The environment variables set specified by the developer are set in the container.
+
+**User Experience 2 – Automatically Injected by platform engineer** 
+
+The platform engineer can specify default environment variables which will automatically be injected into a container when a connection is established. This is similar to existing functionality for portable resource types in Radius today (see the [Redis type](https://docs.radapp.io/reference/resource-schema/cache/redis/#environment-variables-for-connections) for example). Note that user experience 1 and 2 can be used side by side.
```

Comment:
> This is similar to existing functionality for portable resource types in Radius today

Just to be explicit, the key difference will be that the env variable keys will be user defined (example ` POSTGRESQL_CONNECTION_STRING` below) rather than a standard convention decided by Radius.